Wrapper/DBEngine.cpp

```diff
@@ -0,0 +1,273 @@
+#include "QueryEngine/CompilationOptions.h"
+#include "QueryEngine/ResultSet.h"
+#include "QueryRunner/QueryRunner.h"
+#include "Shared/Logger.h"
+#include "Shared/mapdpath.h"
+#include "Shared/sqltypes.h"
+
+namespace EmbeddedDatabase {
+
+enum class ColumnType : uint32_t { eUNK, eINT, eDBL, eFLT, eSTR, eARR };
+
+/**
+ * Cursor internal implementation
+ */
+class CursorImpl : public Cursor {
+ public:
+  CursorImpl(std::shared_ptr<ResultSet> resultSet,
+             std::shared_ptr<Data_Namespace::DataMgr> dataMgr)
+      : m_resultSet(resultSet), m_DataMgr(dataMgr) {}
+
+  size_t GetColCount() { return m_resultSet->colCount(); }
+
+  size_t GetRowCount() { return m_resultSet->rowCount(); }
+
+  Row GetNextRow() {
+    auto row = m_resultSet->getNextRow(true, false);
+    if (row.empty()) {
+      std::cout << "Cursor::GetNextRow - Empty row" << std::endl;
+      return Row();
+    }
+    return Row(row);
+  }
+
+  ColumnType GetColType(uint32_t nPos) {
+    if (nPos < GetColCount()) {
+      SQLTypeInfo typeInfo = m_resultSet->getColType(nPos);
+      switch (typeInfo.get_type()) {
+        case kNUMERIC:
+        case kDECIMAL:
+        case kINT:
+        case kSMALLINT:
+        case kBIGINT:
+          return ColumnType::eINT;
+
+        case kDOUBLE:
+          return ColumnType::eDBL;
+
+        case kFLOAT:
+          return ColumnType::eFLT;
+
+        case kCHAR:
+        case kVARCHAR:
+        case kTEXT:
+          return ColumnType::eSTR;
+
+        default:
+          return ColumnType::eUNK;
+      }
+    }
+    return ColumnType::eUNK;
+  }
+
+ private:
+  std::shared_ptr<ResultSet> m_resultSet;
+  std::weak_ptr<Data_Namespace::DataMgr> m_DataMgr;
+};
+
+/**
+ * DBEngine internal implementation
+ */
+class DBEngineImpl : public DBEngine {
+ public:
+  const int CALCITEPORT = 3279;
+  const std::string OMNISCI_DEFAULT_DB = "omnisci";
+  const std::string OMNISCI_ROOT_USER = "admin";
+  const std::string OMNISCI_DATA_PATH = "//mapd_data";
+
+  void Reset() {
+    // TODO: Destroy all cursors in the m_Cursors
+    if (m_pQueryRunner != nullptr) {
+      m_pQueryRunner->reset();
+    }
+  }
+
+  void ExecuteDDL(const std::string& sQuery) {
+    if (m_pQueryRunner != nullptr) {
+      m_pQueryRunner->runDDLStatement(sQuery);
+    }
+  }
+
+  Cursor* ExecuteDML(const std::string& sQuery) {
+    if (m_pQueryRunner != nullptr) {
+      auto rs = m_pQueryRunner->runSQL(sQuery, ExecutorDeviceType::CPU);
+      m_Cursors.emplace_back(new CursorImpl(rs, m_DataMgr));
+      return m_Cursors.back();
+    }
+    return nullptr;
+  }
+
+  DBEngineImpl(const std::string& sBasePath)
+      : m_sBasePath(sBasePath), m_pQueryRunner(nullptr) {
+    if (!boost::filesystem::exists(m_sBasePath)) {
+      std::cerr << "Catalog basepath " + m_sBasePath + " does not exist.\n";
+      // TODO: Create database if it does not exist
+    } else {
+      MapDParameters mapdParms;
+      std::string sDataPath = m_sBasePath + OMNISCI_DATA_PATH;
```

Comment:
Let's don't use type prefixes for all variables, it looks like wrong code style, at least for omnisci codebase.

Reply:
ok

Reply:
fixed